python/nammu/view/AtfAreaView.py

```diff
@@ -248,12 +248,12 @@ def get_viewport_carets(self):
         and will give the range from the top of the file block to the lower
         extent of the two viewports.
         '''
-        try:
+        if isinstance(self.container, JScrollPane):
             # Assumes a single edit pane
             viewport = self.container.getViewport()
             top_ch, bottom_ch = self.get_viewport_top_bottom(viewport)
 
-        except:
+        else:
```

Comment:
Did you ever find out what was causing the exception (apart from the "expected" error related to the type of `container`)? If we know what it is, and it's not an `AttributeError`, we could keep this as a `try... except` block but only catch `AttributeError`s, and fail in other cases - although I think the type check is also fine here.

Reply:
I do agree that a `try`...`except` may still be useful, but yesterday I tried debugging this issue and it turned out to be a Heisenbug.  Since I made this change I never saw again an error, not even in thread, related to this part of the code :confused: